src/lib.rs

```diff
@@ -4,26 +4,329 @@
+            recipient,
+            account,
+            amount_per_payout,
+            next_payment_at_block,
+            payout_interval,
+            total_reward_received: Zero::zero(),
+            total_reward_missed: Zero::zero(),
+        };
+
+        let relationship_id = Self::next_reward_relationship_id();
+        NextRewardRelationshipId::mutate(|n| {
+            *n += 1;
+        });
+        <RewardRelationships<T>>::insert(relationship_id, relationship);
+        Ok(relationship_id)
+    }
+
+    // Removes a mapping from reward relashionships based on given identifier.
+    pub fn remove_reward_relationship(id: RewardRelationshipId) {
+        <RewardRelationships<T>>::remove(&id);
+    }
+
+    // For reward relationship found with given identifier, new values can be set for
+    // account, payout, block number when next payout will be made and the new interval after
+    // the next scheduled payout. All values are optional, but updating values are combined in this
+    // single method to ensure atomic updates.
+    pub fn set_reward_relationship(
+        id: RewardRelationshipId,
+        account: Option<T::AccountId>,
+        payout: Option<BalanceOf<T>>,
+        next_payment_at: Option<Option<T::BlockNumber>>,
+        payout_interval: Option<Option<T::BlockNumber>>,
+    ) -> Result<(), RewardsError> {
+        ensure!(
+            <RewardRelationships<T>>::exists(&id),
+            RewardsError::RewardRelationshipNotFound
+        );
+        let mut relationship = Self::reward_relationships(&id);
+
+        if let Some(account) = account {
+            relationship.account = account;
+        }
+        if let Some(payout) = payout {
+            relationship.amount_per_payout = payout;
+        }
+        if let Some(next_payout_at_block) = next_payment_at {
+            if let Some(blocknumber) = next_payout_at_block {
+                ensure!(
+                    blocknumber > <system::Module<T>>::block_number(),
+                    RewardsError::BlockNumberInPast
+                );
+            }
+            relationship.next_payment_at_block = next_payout_at_block;
+        }
+        if let Some(payout_interval) = payout_interval {
+            relationship.payout_interval = payout_interval;
+        }
+        <RewardRelationships<T>>::insert(&id, relationship);
+        Ok(())
+    }
+
+    /*
+    For all relationships where next_payment_at_block is set and matches current block height,
+    a call to pay_reward is made for the suitable amount, recipient and source.
+    The next_payment_in_block is updated based on payout_interval.
+    If the call succeeds, total_reward_received is incremented on both
+    recipient and dependency with amount_per_payout, and a call to T::PayoutStatusHandler is made.
+    Otherwise, analogous steps for failure.
+    */
+    fn do_payouts(now: T::BlockNumber) {
+        for relationship_id in FIRST_REWARD_RELATIONSHIP_ID..Self::next_reward_relationship_id() {
```

Comment:
Same comment as in `TokenMint` about `linked_map`. What do you think?

Reply:
yeah using linked_map now